modules/core/src/lib/effect-manager.js

```diff
@@ -63,4 +65,16 @@ export default class EffectManager {
     }
     this.needApplyDefaultLighting = !hasEffect;
```

Comment:
It is unnecessary to attach an additional property to this instance. Just call the method "hasLightingEffect" and return the result.